utils/src/main/java/software/amazon/awssdk/utils/cache/NonBlocking.java

```diff
@@ -61,23 +67,27 @@ public class NonBlocking implements CachedSupplier.PrefetchStrategy {
      * the {@link #EXECUTOR}.
      */
     private static final ScheduledThreadPoolExecutor SCHEDULER =
-        new ScheduledThreadPoolExecutor(1, new ThreadFactoryBuilder().daemonThreads(true).build());
+        new ScheduledThreadPoolExecutor(1, new ThreadFactoryBuilder().threadNamePrefix("sdk-cache-scheduler")
```

Comment:
Perhaps a more specific name: `aws-sdk-credentials-provider-cache-scheduler`, specifying the AWS SDK and that this is a cache for credentials provider (to differentiate from other potential caches in the SDK, now or in the future).

Reply:
The latest revision prepends this to the thread name specified when the prefetch strategy is created, so the complete thread name in logs would be something like sdk-cache-scheduler-0-sts-assume-role-credentials-provider-0. , and idle threads would be sdk-cache-scheduler-0.

Does that make this better?